src/documents/ContextAwareness.md

```diff
@@ -0,0 +1,327 @@
+                localDemoOption: {
+                    contextValue: "{uploader}.options.demo",
+                    gradeNames: "fluid.uploader.demo"
+                }
+            },
+            defaultGradeNames: "fluid.uploader.live"
+        }
+    }
+});
+```
+
+`technology` refers to the implementation technology of the uploader. For previous releases, this supported a Flash-based engine as well as an HTML engine targetted at a back-levelled HTML
+API supported by Firefox 3.x. Although all technologies other than a modern HTML5 engine have been removed from the current framework image, the basic architecture to support other
+engines still exists and could be contributed to in future. The `liveness` adaptation relates to the mocking infrastructure for the Uploader which exists at two levels. Firstly, there is 
+the "demo uploader" which mocks all of the engine-side implementation, and secondly the uploader can be run in various styles of integration tests which only mock the transport level which
+actually performs the file upload.
+
+### Example of dynamically broadcasting a fresh adaptation
+
+Finally, a third dimension of adaptation is supported by the Uploader's capability to be configured in a way that it will respond to previous instances of its own API - in particular that
+delivered for Infusion 1.2, released in April 2010, and Infusion 1.3, released in November 2010. This is implemented by allowing a dynamic contribution of a fresh dimension to the uploader's
+`contextAwareness` record from separate implementation files:
+
+```javascript
+fluid.defaults("fluid.uploader.compatibility.distributor.1_3", {
+    distributeOptions: {
+        record: {
+            "1_2": {
+                contextValue: "{fluid.uploader.requiredApi}.options.value",
+                equals: "fluid_1_2",
+                gradeNames: "fluid.uploader.compatibility.1_2"
+            }
+        },
+        target: "{/ fluid.uploader}.options.contextAwareness.apiCompatibility.checks"
+    }
+});
+
+// Actually construct a component instance performing the options broadcast into all uploaders
+fluid.constructSingle([], {
+    singleRootType: "fluid.uploader.compatibility.distributor",
+    type: "fluid.uploader.compatibility.distributor.1_3"
+});
+
+// The grade that contextAwareness ends up contributing into the uploader, if the rule is activated
+fluid.defaults("fluid.uploader.compatibility.1_2", {
+    transformOptions: {
+        transformer: "fluid.model.transformWithRules",
+        config: fluid.compat.fluid_1_2.uploader.optionsRules
+    }
+});
+```
+
+The defaults block `fluid.uploader.compatibility.distributor.1_3` contains an options distribution which appeals to the existence of a third dimension in the uploader's `contextAwareness`, named
+`apiCompatibility` &#8212; this can be done simply by arranging to broadcast the appropriate options into it. Once we have defined this options distribution, we actually need to construct a component
```

Comment:
Is this example of constructing a singleton instance a part of working with the contextAwareness api? Is all use of this api going to involve this? If not, then I think the example is introducing some unrelated and thus confusing things that cloud the issue. It would be better to have a more directly relevant example.